### ISSUE TITLE
Timed out waiting for authentication provider to register

### ISSUE THREAD

**Issue by zebleck** [microsoft/vscode-azure-account#897]
Does this occur consistently? Yes
Repro steps:
Install Azure Tools extensions including Azure account in `Cursor` Editor.
Switch to Azure Tab.

Error: Timed out waiting for authentication provider to register

Version: 0.11.6
OS: linux
OS Release: 5.4.72-microsoft-standard-WSL2
Product: Cursor
Product Version: 1.84.2
Language: en

**Comment by 00MB** [comment]
Getting this as well

**Comment by zebleck** [comment]
nope still happening for me.

**Comment by alexweininger** [comment]
Are you all experiencing this issue using the Cursor editor like the original issue states? Or using VS Code?

**Comment by TomMoeras** [comment]
I'm having a similar issue connecting to a remote tunnel, same error message (using Cursor, works fine in VS Code). I suspect that it is related to how Cursor handles this somehow.

**Comment by Raphael9999** [comment]
I had a similar issue and received assistance from MS. Here is how we solved it: 

- On the left bar, Select Extensions (Ctrl Shift X)
- In the search bar at the top of the extension pane, enter: "@builtin Microsoft"
- When doing that, the "Microsoft Account" extension appears. Mine was disabled. 
- Click the extension and enable it (there is a little button in the main pane for that). 
- Restart Visual Studio Code,

Because I am working with Sovereign Cloud, I also had to 
- Go File > Preferences > Settings, look for "Microsoft-sovereign-cloud.environment", pick yours. 
- Open the command palette (Ctrl Shift P) and run "Azure: Signin azureResourceGroups/logIn"


I hope this helps.

**Comment by marcosclub** [comment]
Yep, experiencing the same, here. Pity...

**Comment by JislaLifeBox** [comment]
Experiencing the same

**Comment by seo1623** [comment]
I have implemented all the advice written here, but the same error continues. I might give up on the cursor.

**Comment by port2077** [comment]
Did anyone find any solution to this ?

**Comment by mohitmehral** [comment]
getting same on cursor IDE however working fine in vscode

**Comment by duytrand** [comment]
any fix on this? still failing. i don't see the fix anywhere

**Comment by fitchmultz** [comment]
Found the fix elsewhere. https://forum.cursor.com/t/azure-extension-doesent-work/3949/15

**Fix Azure extensions on Cursor**
Cursor does not come with Microsoft authentication, a builtin extension for vscode

To get this extension, grab it from VSCode
`\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\microsoft-authentication`
Copy to
`\AppData\Local\Programs\cursor\resources\app\extensions`

For MacOS

`/Applications/Visual Studio Code. app/Contents/Resources/app/extensions/microsoft-authentication`
and
`/Applications/Cursor.app/Contents/Resources/app/extensions`

Reload Cursor windows after. 

See additional steps at the link to connect to servers.

@duytrand @mohitmehral @marcosclub @port2077 @TomMoeras @seo1623 @mat-ej

**Comment by mllopart** [comment]
> Found the fix elsewhere. https://forum.cursor.com/t/azure-extension-doesent-work/3949/15
> 
> **Fix Azure extensions on Cursor** Cursor does not come with Microsoft authentication, a builtin extension for vscode
> 
> To get this extension, grab it from VSCode `\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\microsoft-authentication` Copy to `\AppData\Local\Programs\cursor\resources\app\extensions`
> 
> For MacOS
> 
> `/Applications/Visual Studio Code. app/Contents/Resources/app/extensions/microsoft-authentication` and `/Applications/Cursor.app/Contents/Resources/app/extensions`
> 
> Reload Cursor windows after.
> 
> See additional steps at the link to connect to servers.
> 
> [@duytrand](https://github.com/duytrand) [@mohitmehral](https://github.com/mohitmehral) [@marcosclub](https://github.com/marcosclub) [@port2077](https://github.com/port2077) [@TomMoeras](https://github.com/TomMoeras) [@seo1623](https://github.com/seo1623) [@mat-ej](https://github.com/mat-ej)

This has solved all the issues! Thanks @fitchmultz

**Comment by duytrand** [comment]
it works, but i notice that the cursor keeps deleting (or i have to keep
recopy that ) the location.


*Duy Tran *
Google - Cloud Solution Architect
PropLytics - Chief Executive Officer
Email:  ***@***.***
Linkedin <https://www.linkedin.com/in/duytranduc/> | Website
<https://itsduy.com/>

**Comment by duytrand** [comment]
just realized, everytime cursor updates, it get deleted, so i have to
recopy


*Duy Tran *
Google - Cloud Solution Architect
PropLytics - Chief Executive Officer
Email:  ***@***.***
Linkedin <https://www.linkedin.com/in/duytranduc/> | Website
<https://itsduy.com/>

**Comment by fitchmultz** [comment]
I created a simple shell script and keep it in my main projects directory.
After each update I just run the script again.

Mitch Fultz
***@***.***

**Comment by PierceLovesee** [comment]
Thank you, @fitchmultz, for the fix!!

**Comment by Dbanks39** [comment]
> Does this occur consistently? Yes Repro steps: Install Azure Tools extensions including Azure account in `Cursor` Editor. Switch to Azure Tab.
> 
> Error: Timed out waiting for authentication provider to register
> 
> Version: 0.11.6 OS: linux OS Release: 5.4.72-microsoft-standard-WSL2 Product: Cursor Product Version: 1.84.2 Language: en

getting the same issue when attempting to use the latest version of cursor with azure resources extensions

**Comment by fitchmultz** [comment]
You have to repeat the process after every update to cursor unfortunately

**Comment by postcode-ai** [comment]
TL;DR: 

**Cursor is missing the VSCode microsoft-authentication extension which for the mac _should_ be located at**

**/Applications/Visual Studio Code. app/Contents/Resources/app/extensions/microsoft-authentication**

(if you've installed vscode ofc)

**If you can't find the microsoft-authentication extension** on a mac at the path above then, you can find it in a VS Code install download, and you don't have to install or re-install VS Code.

1. Download VS Code. - currently at https://code.visualstudio.com/docs/?dv=osx 
2. Use FInder to go to the download and right click on the newly downloaded "Visual Studio Code.app" file - image below


<img width="556" alt="Image" src="https://github.com/user-attachments/assets/1c54a6b0-dcc5-4f49-8206-ca3154f32724" />

#

3. Click on "Show Package Contents"
4. Use Finder to go to '~/Downloads/Visual Studio Code.app/Contents/Resources/app/extensions/'
#

<img width="727" alt="Image" src="https://github.com/user-attachments/assets/0e43d9ae-fc04-4da8-bd6a-02e57cee15eb" />

#
5. Right-click on the "microsoft-authentication" folder and choose "Copy"
#

<img width="724" alt="Image" src="https://github.com/user-attachments/assets/7776306d-9371-46d5-ae14-59fcf3a30e97" />

#

6. Use Finder to go to /Applications/**Cursor.app**/Contents/Resources/app/extensions/
#

<img width="506" alt="Image" src="https://github.com/user-attachments/assets/e599f574-ade0-4023-93e3-d144289da4ec" />
#

7. Paste the copied "microsoft-authentication" folder in there
8. **Fully exit Cursor and re-open it**
9. You should now be able to connect to Azure using he eAzure extension or whatever you use


_N.B. There are instructions for **windows paths** further up in this thread but. I haven't tested them so try them out ._
